### PR TITLE
Add MarkovChain DOT export

### PR DESCRIPTION
**Pull request by jschueller**
from the test (I the meantime modified the code for nodes numbering to start from 1):
markov

### CODE REVIEW

File: lib/src/Uncertainty/Process/DiscreteMarkovChain.cxx [hunk 1]
```diff
@@ -294,6 +294,22 @@ void DiscreteMarkovChain::setMesh(const Mesh & mesh)
...
+{
+  std::ofstream dotFile(filename.c_str());
+  if (dotFile.fail())
+    throw FileOpenException(HERE) << "Could not open file " << filename;
+  dotFile.imbue(std::locale("C"));
+  dotFile << "digraph " << getName() << " {\n";
+  for (UnsignedInteger j = 0; j < transitionMatrix_.getNbRows(); ++ j)
+    for (UnsignedInteger i = 0; i < transitionMatrix_.getNbRows(); ++ i)
+      if (transitionMatrix_(i, j) > 0.0)
+        dotFile << j+1 << "->" << i+1 << " [label=" << transitionMatrix_(i, j) << "]\n";
```
Comment (regislebrun) [inline lib/src/Uncertainty/Process/DiscreteMarkovChain.cxx]: It should be `dotFile << j << "->" << i << " [label=" << transitionMatrix_(i, j) << "]\n";` as the state space is `[[0,...,n-1]]` and not `[[1,...,n]]`.
What's about using some OT-ish colors? Something like:
```
  dotFile << "layout=circo\n";
  dotFile << "resolution=300\n";
  dotFile << "node[shape=circle, color=red]";
  for (UnsignedInteger j = 0; j < transitionMatrix_.getNbRows(); ++ j)
    for (UnsignedInteger i = 0; i < transitionMatrix_.getNbRows(); ++ i)
      if (transitionMatrix_(i, j) > 0.0)
        dotFile << j+1 << "->" << i+1 << " [label=" << transitionMatrix_(i, j) <<  ", weight=0.5, fontsize=12, color=green]\n";
```

Comment (regislebrun) [reply lib/src/Uncertainty/Process/DiscreteMarkovChain.cxx]: Anyway, good job!

Comment (jschueller) [reply lib/src/Uncertainty/Process/DiscreteMarkovChain.cxx]: it's a bit too much for my eyes : )
markov2

Comment (jschueller) [reply lib/src/Uncertainty/Process/DiscreteMarkovChain.cxx]: But I added resourcemap keys to customize the graph settings

Comment (regislebrun) [reply lib/src/Uncertainty/Process/DiscreteMarkovChain.cxx]: You don't like pizzas ;-)?

**Comment by regislebrun** [comment]
+1

**Pull request merged**